connectivity/FEATURE_BLE/include/ble/gatt/GattCharacteristic.h

```diff
@@ -1750,16 +1756,40 @@ class GattCharacteristic {
      *
      * @return A pointer the requested descriptor if @p index is within the
      * range of the descriptor array or nullptr otherwise.
+     *
+     * @note if this characteristic has an implicitly-created CCCD this
+     * descriptor will be available at the highest index
+     * (ie: return of getDescriptorCount() - 1)
      */
     GattAttribute *getDescriptor(uint8_t index)
     {
-        if (index >= _descriptorCount) {
+
+        if(index == _descriptorCount) {
+            // If _implicit_cccd is nullptr, we want to return that anyway
+            return _implicit_cccd;
+        }
+        else if (index > _descriptorCount) {
             return nullptr;
         }
 
         return _descriptors[index];
     }
 
+    /**
+     * Sets this GattCharacteristic's implicitly-created CCCD, if
+     * applicable.
+     *
+     * @note once this is called, the pointed-to GattAttribute
+     * is owned by this GattCharacteristic and will be deleted
+     * during this object's destructor
+     *
+     * @note this is an internal function and should not be called
```

Comment:
then it should be private

Reply:
Haha, didn't read your comment about "I wasn't able to get it to compile due to all the weird namespacing stuff going on with ble::GattServer and ble::impl::GattServer -- it boggled my mind"
Don't worry I have the same experience (even confuses IDEs).

Reply:
I have added this doxygen preprocessor guard, resolved by fa2c363